packages/discovery/package.json

```diff
@@ -0,0 +1,77 @@
+{
+  "name": "@web3-storage/discovery",
+  "version": "1.0.0",
+  "description": "protocol for content discovery",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/web3-storage/w3up.git",
+    "directory": "packages/discovery"
+  },
+  "license": "(Apache-2.0 OR MIT)",
+  "type": "module",
+  "types": "dist/src/index.d.ts",
+  "main": "src/index.js",
+  "files": [
+    "src",
+    "test",
+    "dist/**/*.d.ts",
+    "dist/**/*.d.ts.map"
+  ],
+  "exports": {
+    ".": {
+      "types": "./dist/src/index.d.ts",
+      "import": "./src/index.js"
+    }
+  },
+  "scripts": {
+    "build": "tsc --build",
+    "check": "tsc --build",
+    "lint": "tsc --build",
+    "prepare": "npm run build",
+    "prettier:check": "prettier --check '{src,test}/*.{js,ts,yml,json}'",
+    "test": "mocha --bail --timeout 10s -n no-warnings -n experimental-vm-modules -n experimental-fetch test/**/*.spec.js",
+    "test-watch": "pnpm build && mocha --bail --timeout 10s --watch --parallel -n no-warnings -n experimental-vm-modules -n experimental-fetch --watch-files src,test"
+  },
+  "devDependencies": {
+    "@types/mocha": "^10.0.1",
+    "@types/node": "^18.11.18",
+    "@typescript-eslint/eslint-plugin": "^5.58.0",
+    "@typescript-eslint/parser": "^5.58.0",
+    "eslint": "^8.38.0",
+    "eslint-plugin-jsdoc": "^41.1.1",
+    "mocha": "^10.2.0",
+    "typescript": "^5.0.4"
```

Comment:
This could be causing problems with TS as ucanto is still on [4.9.x](https://github.com/web3-storage/ucanto/blob/main/packages/server/package.json#L50) and there may have been a change in TS that affects inference.